src/pathfinder.rs

```diff
@@ -0,0 +1,356 @@
+use num::integer::sqrt;
+use pathfinding::prelude::astar;
+use serde::{Deserialize, Serialize};
+use std::borrow::Borrow;
+use std::hash::Hash;
+use std::num::ParseIntError;
+use std::sync::{Arc, RwLock};
+use std::usize::MAX;
+
+// static mut are not thread safe, and manipulating one is only possible behind unsafe tag
+// However, most of the unsafe call should be thread safe because nodes inside this vec are behind a RwLock
+static mut NODES: Vec<NodeContainer> = Vec::new();
+
+// Container for a node. Exist mainly to be able to implement Hash, which is not implemented for RefCell
```

Comment:
I don't really know why Hash isn't implemented for RefCell, anyone knows why ?

Reply:
HashMaps play badly (not UB, but unspecified) if the hashes change while the object is inside. Interior mutability makes that trivial.